confignode/src/main/java/org/apache/iotdb/confignode/manager/load/LoadManager.java

```diff
@@ -179,6 +158,51 @@ public void run() {
     }
   }
 
+  private THeartbeatReq genHeartbeatReq() {
+    return new THeartbeatReq(System.currentTimeMillis());
+  }
+
+  private void doLoadBalancing(int balanceCount) {
+    if (balanceCount % 5 == 0) {
+      // We update nodes' load statistic in every 5s
+      updateNodeLoadStatistic();
+    }
+  }
+
+  private void updateNodeLoadStatistic() {
+    heartbeatCacheMap.values().forEach(IHeartbeatStatistic::updateLoadStatistic);
+  }
+
+  /**
+   * Send heartbeat requests to all the online DataNodes
+   *
+   * @param onlineDataNodes DataNodes that currently online
+   */
+  private void pingOnlineDataNodes(List<TDataNodeInfo> onlineDataNodes) {
+    // Discard heartbeat cache of the offline DataNodes
+    Set<Integer> onlineIds = new HashSet<>();
+    for (TDataNodeInfo dataNodeInfo : onlineDataNodes) {
+      onlineIds.add(dataNodeInfo.getLocation().getDataNodeId());
+    }
+    for (Integer dataNodeId : heartbeatCacheMap.keySet()) {
+      if (!onlineIds.contains(dataNodeId)) {
+        heartbeatCacheMap.remove(dataNodeId);
+      }
+    }
```

Comment:
The cache could be remove when removing datanode. We do not discard cache here.

Reply:
Fixed.